pynetbox_data_uploader/tests/test_csv_to_netbox.py

```diff
@@ -0,0 +1,283 @@
+from unittest.mock import patch
+from dataclasses import asdict
+from pytest import fixture, raises
+from lib.user_methods.csv_to_netbox import (
+    CsvToNetbox,
+    arg_parser,
+    do_csv_to_netbox,
+    main,
+)
+from lib.utils.device_dataclass import Device
+
+
+# Two tests mock the same Device dataclass and therefore have duplicate code.
+# pylint: disable=R0801
+@fixture(name="instance")
+def instance_fixture():
+    """
+    This method calls the class being tested.
+    :return: The class with mock arguments.
+    """
+    mock_url = "mock_url"
+    mock_token = "mock_token"
+    return CsvToNetbox(mock_url, mock_token)
+
+
+def test_check_file_path_valid(instance):
+    """
+    This test checks that the method doesn't raise any errors for valid filepaths.
+    """
+    mock_file_path = "."
+    instance.check_file_path(mock_file_path)
+
+
+def test_check_file_path_invalid(instance):
+    """
+    This test checks that the method raises an error when the filepath is invalid.
+    """
+    mock_file_path = ">"
+    with raises(FileNotFoundError):
+        instance.check_file_path(mock_file_path)
+
+
+@patch("lib.user_methods.csv_to_netbox.open_file")
+@patch("lib.user_methods.csv_to_netbox.separate_data")
+def test_read_csv(mock_separate_data, mock_open_file, instance):
+    """
+    This test ensures the file open method is called and the separate data method is called.
+    """
+    mock_file_path = "mock_file_path"
+    res = instance.read_csv(mock_file_path)
+    mock_open_file.assert_called_once_with(mock_file_path)
+    mock_separate_data.assert_called_once_with(mock_open_file.return_value)
+    assert res == mock_separate_data.return_value
+
+
+@patch("lib.user_methods.csv_to_netbox.NetboxCheck.check_device_exists")
+def test_check_netbox_device_does_exist(mock_check_device_exists, instance):
+    """
+    This test ensures that an error is raised if a device does exist in Netbox.
+    """
+    device = Device(
+        tenant="t1",
+        device_role="dr1",
+        manufacturer="m1",
+        device_type="dt1",
+        status="st1",
+        site="si1",
+        location="l1",
+        rack="r1",
+        face="f1",
+        airflow="a1",
+        position="p1",
+        name="n1",
+        serial="se1",
+    )
```

Comment:
looks like you're defining this multiple times in different tests - consider using a fixture to remove duplication

Reply:
👍